### PR TITLE
Fixes #5691. Support gitlab forking_access_level, builds_access_level and container_registry_access_level fields

### LINKED ISSUE #5691: Support gitlab forking_access_level, builds_access_level and container_registry_access_level fields
### Summary

Add support for the following gitlab project fields (see [gitlab api rest](https://docs.gitlab.com/ee/api/projects.html#create-project) for detail of these fields):
- `forking_access_level`: shown as `Repository forks`  project setting in gitlab web user interface
- `builds_access_level`: shown as `Repository CI/CD` project setting in gitlab web user interface
- `container_registry_access_level`:  shown as `Container registry`  project setting in gitlab web user interface

### Issue Type

Feature Idea

### Component Name

gitlab_project

### Additional Information

Example of project creation using the new fields:

```yaml
- name: Setup gitlab project
  community.general.gitlab_project:
    api_url: https://gitlab.example.com/
    validate_certs: true
    api_username: dj-wasabi
    api_password: "MySecretPassword"
    name: my_project
    group: ansible
    builds: disabled
    container_registry: disabled
    forks: disabled
    state: present
  delegate_to: localhost
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by juanvalino**
##### SUMMARY

Fixes #5691

Added support for `gitlab forking_access_level`, `builds_access_level` and `container_registry_access_level` fields (see [gitlab api rest](https://docs.gitlab.com/ee/api/projects.html#create-project) for detail of these fields).

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
gitlab_project

##### ADDITIONAL INFORMATION
Example of project creation using the new fields:

```yaml
- name: Setup gitlab project
  community.general.gitlab_project:
    api_url: https://gitlab.example.com/
    validate_certs: true
    api_username: dj-wasabi
    api_password: "MySecretPassword"
    name: my_project
    group: ansible
    builds: disabled
    container_registry: disabled
    forks: disabled
    state: present
  delegate_to: localhost
```

### CODE REVIEW

File: plugins/modules/gitlab_project.py [hunk 1]
```diff
@@ -172,6 +172,39 @@
...
+    description:
+      - C(private) Repository CI/CD allowed only to project members.
+      - C(disabled) Repository CI/CD is disabled.
+      - C(enabled) Repository CI/CD is enabled.
+    default: enabled
+    type: str
+    choices: ["private", "disabled", "enabled"]
+    aliases:
+      - builds_access_level
+    version_added: "6.1.0"
```
Comment (felixfontein) [inline plugins/modules/gitlab_project.py:185]: 6.1.0 has already been released, you cannot add new options for it. The next release will be 6.2.0:
```suggestion
    version_added: "6.2.0"
```
(same below)

File: plugins/modules/gitlab_project.py [hunk 2]
```diff
@@ -172,6 +172,39 @@
       - This option is only used on creation, not for updates. This is also only used if I(initialize_with_readme=true).
     type: str
     version_added: "4.2.0"
+  builds:
+    description:
+      - C(private) Repository CI/CD allowed only to project members.
+      - C(disabled) Repository CI/CD is disabled.
+      - C(enabled) Repository CI/CD is enabled.
+    default: enabled
```
Comment (felixfontein) [inline plugins/modules/gitlab_project.py:180]: Default values are dangerous. There should not be a default, and the module should only change the value if the user explicitly specified something.

File: plugins/modules/gitlab_project.py [hunk 3]
```diff
@@ -172,6 +172,39 @@
       - This option is only used on creation, not for updates. This is also only used if I(initialize_with_readme=true).
     type: str
     version_added: "4.2.0"
+  builds:
+    description:
+      - C(private) Repository CI/CD allowed only to project members.
```
Comment (felixfontein) [inline plugins/modules/gitlab_project.py:177]: Please write a complete sentence in every paragraph:
```suggestion
      - C(private) means that repository CI/CD allowed only to project members.
```

File: plugins/modules/gitlab_project.py [hunk 4]
```diff
@@ -172,6 +172,39 @@
...
+      - C(disabled) Repository CI/CD is disabled.
+      - C(enabled) Repository CI/CD is enabled.
+    default: enabled
+    type: str
+    choices: ["private", "disabled", "enabled"]
+    aliases:
+      - builds_access_level
```
Comment (felixfontein) [inline plugins/modules/gitlab_project.py:184]: Aliases should be avoided for new options, unless they are absolutely necessary. What is the reason for having an alias here?

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution. I've added some first comments. Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

File: plugins/modules/gitlab_project.py (hunk 2 above)
Comment (juanvalino) [reply plugins/modules/gitlab_project.py:180]: Ok. I'll remove defaults

File: plugins/modules/gitlab_project.py (hunk 4 above)
Comment (juanvalino) [reply plugins/modules/gitlab_project.py:184]: The aliases were introduced to clarify the gitlab api rest attributed managed, but you're right. I'll remove the aliases.

Comment (nejch) [reply plugins/modules/gitlab_project.py]: Personally I think it'd be better if the attribute keys follow the GitLab API ones anyway, just as all of the other ones do.

Users will have to consult the upstream GitLab API docs in any case for some of these, even just to check the permissions or the type of token required. So it makes it easier to map to this module IMO.

**Comment by juanvalino** [comment]
I'll close this PR and open a new on will all changes consolidated.

File: plugins/modules/gitlab_project.py (hunk 4 above)
Comment (juanvalino) [reply plugins/modules/gitlab_project.py]: OK

**Pull request closed without merging**